### PR TITLE
Support salesforce data source

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

To Support salesforce data source, introduce the below APIs:

- loginToSalesforce
- querySalesforceData
- querySaleseforceMetadata
- querySaleseforceTableDetails

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/salesforce.R [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#' API to login to Salesforce
+#' export
+#' @param sever - (optional) login server
+#' @param username - Salesforce login user name
+#' @param password - Salesforce login password
+#' @param securityToken - (optional) security token if required
+#' This API
```
Comment (hideaki) [inline R/salesforce.R:7]: Clean this comment line.

File: R/salesforce.R [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+#' @param sever - (optional) login server
+#' @param username - Salesforce login user name
+#' @param password - Salesforce login password
+#' @param securityToken - (optional) security token if required
+querySaleseforceMetadata <- function(server = NULL, username, password, securityToken = NULL){
+  if (!requireNamespace("salesforcer")) {
+    stop("package salesforcer must be installed.")
+  }
+  loginToSalesforce(server = server, username = username, password = password, securityToken = securityToken)
+  my_queries <- list(list(type='CustomObject'))
```
Comment (hideaki) [inline R/salesforce.R:33]: Does this query include query for standard objects?

File: R/salesforce.R (hunk 1 above)
Comment (hidekoji) [reply R/salesforce.R]: Updated the comment.

File: R/salesforce.R (hunk 2 above)
Comment (hidekoji) [reply R/salesforce.R:35]: Added comment.

**Pull request merged**